sys/aarch64/rootdev.c

```diff
@@ -290,22 +290,22 @@ static void rootdev_deactivate_resource(device_t *dev, res_type_t type,
   /* TODO: unmap mapped resources. */
 }
 
-static bus_driver_t rootdev_driver = {
-  .driver =
-    {
-      .size = sizeof(rootdev_t),
-      .desc = "RPI3 platform root bus driver",
-      .attach = rootdev_attach,
-    },
-  .bus =
-    {
-      .intr_setup = rootdev_intr_setup,
-      .intr_teardown = rootdev_intr_teardown,
-      .alloc_resource = rootdev_alloc_resource,
-      .release_resource = rootdev_release_resource,
-      .activate_resource = rootdev_activate_resource,
-      .deactivate_resource = rootdev_deactivate_resource,
-    },
+static bus_methods_t rootdev_bus_if = {
+  .intr_setup = rootdev_intr_setup,
+  .intr_teardown = rootdev_intr_teardown,
+  .alloc_resource = rootdev_alloc_resource,
+  .release_resource = rootdev_release_resource,
+  .activate_resource = rootdev_activate_resource,
+  .deactivate_resource = rootdev_deactivate_resource,
+};
+
+static driver_t rootdev_driver = {
+  .size = sizeof(rootdev_t),
+  .desc = "RPI3 platform root bus driver",
+  .attach = rootdev_attach,
+  .interfaces = {
+    [DIF_BUS] = &rootdev_bus_if,
```

Comment:
NIT: I am not sure if it is better to add functions like:
```
____ bus_register_interface(driver_t drv)
____ pci_bus_register_interface(driver_t drv)
```
and to register that interfaces in driver's attach function - I think it is more linux-style

Reply:
FTTB I don't think I can come up with use case for such interface.